app/code/community/Algolia/Algoliasearch/Helper/Entity/Producthelper.php

```diff
@@ -905,15 +905,16 @@ public function getObject(Mage_Catalog_Model_Product $product)
             if ($attribute_resource) {
                 $attribute_resource->setStoreId($product->getStoreId());
 
+                $values = array();
+                $subProductImages = array();
+
                 /**
                  * if $value is missing or if the attribute is SKU,
                  * use values from child products.
                  */
                 if (($value === null || 'sku' == $attribute_name) && ($type == 'configurable' || $type == 'grouped' || $type == 'bundle')) {
-                    if ($value === null) {
-                        $values = array();
-                    } else {
-                        $values = array($this->getValueOrValueText($product, $attribute_name, $attribute_resource));
+                    if ($value !== null) {
```

Comment:
I think this condition will always be true since we define `$values = array();` line 908.